File: src/main/kotlin/net/merayen/elastic/backend/architectures/local/nodes/mix_1/LNode.java
package net.merayen.elastic.backend.architectures.local.nodes.mix_1;

import java.util.Map;

import net.merayen.elastic.backend.architectures.local.LocalNode;
import net.merayen.elastic.backend.architectures.local.LocalProcessor;

public class LNode extends LocalNode {
	float mix;
	public LNode() {
		super(LProcessor.class);
	}

	@Override
	protected void onInit() {}

	@Override
	protected void onSpawnProcessor(LocalProcessor lp) {}

	@Override
	protected void onProcess(Map<String, Object> data) {}

	@Override
	protected void onParameter(String key, Object value) {
		if(key.equals("mix"))
			mix = ((Number)value).floatValue();
	}

	@Override
	protected void onFinishFrame() {}

	@Override
	protected void onDestroy() {}
}

File: src/main/kotlin/net/merayen/elastic/backend/architectures/local/nodes/midi_transpose_1/LNode.kt
package net.merayen.elastic.backend.architectures.local.nodes.midi_transpose_1

import net.merayen.elastic.backend.architectures.local.LocalNode
import net.merayen.elastic.backend.architectures.local.LocalProcessor

class LNode : LocalNode(LProcessor::class.java) {
	var transpose = 0

	override fun onInit() {}
	override fun onSpawnProcessor(lp: LocalProcessor?) {}
	override fun onProcess(data: MutableMap<String, Any>?) {}

	override fun onParameter(key: String?, value: Any?) {
		if (key == "transpose" && value is Number)
			transpose = value.toInt()
	}

	override fun onFinishFrame() {}
	override fun onDestroy() {}
}